controllers/events.js

```diff
@@ -259,6 +260,35 @@ const kickoutPeer = async (req, res) => {
   }
 };
 
+const generateEventCode = async (req, res) => {
+  // this id is for events
+  const { id } = req.params;
+  const { eventCode, role } = req.body;
+  const eventCodeUuid = crypto.randomUUID({ disableEntropyCache: true });
+
+  if (role !== ROLES.MAVEN) {
+    return res.status(400).json({
+      message: "Currently the room codes feature is only for mavens!",
+    });
+  }
+
+  try {
+    const eventCodeObjectFromDB = await eventQuery.createEventCode({
+      id: eventCodeUuid,
+      event_id: id,
+      code: eventCode,
+      role,
+    });
```

Comment:
It only gives one code at a time which is not correct. As one room can have multiple codes.